storage/replica_test.go

```diff
@@ -577,6 +649,34 @@ func TestRangeTSCacheLowWaterOnLease(t *testing.T) {
 	}
 }
 
+// TestRangeLeaderLeaseRejectUnknownRaftNodeID ensures that a replica cannot
+// obtain the leader lease if it is not part of the current range descriptor.
+func TestRangeLeaderLeaseRejectUnknownRaftNodeID(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	tc.manualClock.Increment(int64(DefaultLeaderLeaseDuration + 1))
+	now := tc.clock.Now()
+	lease := &proto.Lease{
+		Start:      now,
+		Expiration: now.Add(10, 0),
+		RaftNodeID: proto.MakeRaftNodeID(2, 2),
+	}
+	args := &proto.BatchRequest{}
+	args.Add(&proto.LeaderLeaseRequest{Lease: *lease})
+	errChan, pendingCmd := tc.rng.proposeRaftCommand(tc.rng.context(), args)
```

Comment:
Yeah, this is fine if you need to hook everything up properly. I was hoping that there would be some shortcut but if not this is fine. `client_raft_test.go` might be a better place for this test since it has all of that machinery.